src/events/messageDelete.ts

```diff
@@ -25,46 +23,76 @@ export default async function messageDeleteHandler(message: Message | PartialMes
 		channelName = message.channel.name;
 	}
 
+	// * Audit logs will not show up at the same time the message delete event
+	await new Promise(resolve => setTimeout(resolve, 1000));
+
+	const auditLogs = await guild.fetchAuditLogs({
+		type: AuditLogEvent.MessageDelete,
+		limit: 5
+	});
+	const auditLogEntry = auditLogs.entries.find(entry =>
+		entry.target.id === message.author.id &&
+		entry.extra.channel.id === message.channelId &&
+		Date.now() - entry.createdTimestamp < 5 * 60 * 1000
+	);
+	// * This large time range is needed because message delete audit log entries are combined if multiple deletions are
+	// * performed with the same target user, executor, and channel within 5 minutes. Decreasing this time range would
+	// * create a risk of false negatives because the combined entry keeps the timestamp of the first deletion.
+	// ! Edge case: If a user deletes their own message after a moderator deletes one of theirs within 5 minutes and in
+	// ! the same channel, there will be a false positive mod log entry.
+
+	const isDeletedByModerator = auditLogEntry !== undefined;
```

Comment:
I do agree with your original message about this not being super great. This looks like it would be very error prone in large servers like ours where tons of messages are deleted all the time. I'm not sure this is worth keeping

Reply:
I've gone back and forth on this. I think that it is worth trying out, but if the edge case is hit often enough that it causes a significant amount of confusion, it should be removed. I don't know how common it is for users to delete their own messages within 5 minutes of one of theirs getting deleted by a mod.